src/Microsoft.AspNet.OData.Shared/DefaultContainerBuilder.cs

```diff
@@ -82,6 +82,25 @@ public class DefaultContainerBuilder : IContainerBuilder
         /// </summary>
         /// <returns>The container built by this builder.</returns>
         public virtual IServiceProvider BuildContainer()
+        {
+            try
+            {
+                // On .NET Framework platform, if using Microsoft.Extensions.DependencyInjection > 6.x version,
+                // The runtime throws exception directly when calling BuildContainer if we can 'services.BuildServiceProvider()'
```

Comment:
Yes. the runtime throws an exception directly when calling BuildContainer() if we call "BuildServiceProvider" directly. I don't know the root cause, it could be related to the C# runtime.  Now, if we put it into a private method call, we can catch the exception now.

Actually, yes, we can just do try catch in the caller. Let me verify it.